app/packs/src/decidim/vocdoni/admin/steps/vote.js

```diff
@@ -1,41 +1,67 @@
-const fetchTheVotesStats = async () => {
-  const electionVotesMetadataTable = document.querySelector(ELECTION_VOTES_SELECTOR);
-  if (!electionVotesMetadataTable) {
-    return;
+// Update the DOM with the answer for a specific question index
+const updateDomWithAnswer = (questionIdx, answer, value) => {
+  const dom = document.querySelector(`td[data-question-idx="${questionIdx}"][data-answer-value="${value}"]`);
+  if (dom) {
+    dom.innerHTML = answer;
   }
+};
+
+// Fetch and display vote statistics for a given question index
+const fetchAndDisplayVotes = async (questionIdx) => {
+  const { election } = await getElectionResults();
+  election[questionIdx]?.forEach(updateDomWithAnswer.bind(null, questionIdx));
+};
 
-  // Fetch the votes stats from the Vocdoni API
-  const fetchVotesStats = async () => {
-    const { election } = await getElectionResults();
-    console.log("ELECTION METADATA => ", election);
-    console.group("Partial Results");
-    election.forEach((question, idx) => {
-      question.forEach((answer, value) => {
-        const dom = document.querySelector(`td[data-question-idx="${idx}"][data-answer-value="${value}"]`);
-        console.log(`Question ${idx} - ANSWER ${value} VOTES ${answer}`, dom);
-        dom.innerHTML = answer;
-      });
-    });
-    console.groupEnd();
-  };
+// Process each accordion element to fetch and display stats if it's open
+const processAccordion = async (accordion) => {
+  if (accordion.style.display !== "none") {
+    const questionIdxCell = accordion.querySelector("td[data-question-idx]");
+    if (questionIdxCell) {
+      const questionIdx = questionIdxCell.dataset.questionIdx;
+      await fetchAndDisplayVotes(questionIdx);
+    }
+  }
+};
 
-  fetchVotesStats();
-  setInterval(fetchVotesStats, WAIT_TIME_MS);
+// Check and fetch statistics for each accordion content on DOMContentLoaded
+const checkAndFetchStats = async () => {
+  document.querySelectorAll(".accordion-content").forEach(processAccordion);
 };
 
+// Set up initial fetch for all open accordions on page load
 document.addEventListener("DOMContentLoaded", () => {
-  fetchTheVotesStats();
+  checkAndFetchStats().then(() => console.log("Initial stats fetched"));
+  setInterval(checkAndFetchStats, WAIT_TIME_MS);
+});
+
+// Set up event listeners on accordion titles to fetch stats when they are opened
+document.querySelectorAll(".accordion-title").forEach((title) => {
+  title.addEventListener("click", async (event) => {
+    const accordionItem = event.target.closest(".accordion-item");
+    const accordionContent = accordionItem.querySelector(".accordion-content");
+    const isExpanded = accordionContent.getAttribute("aria-expanded") === "true";
+    if (!isExpanded) {
+      const questionIdxCell = accordionContent.querySelector("td[data-question-idx]");
+      if (questionIdxCell) {
+        const questionIdx = questionIdxCell.dataset.questionIdx;
+        await fetchAndDisplayVotes(questionIdx);
+      }
+    }
```

Comment:
These two condictionals can be refactored to use a guard clause. I think it'd be cleaner